src/etc/inc/openvpn.inc

```diff
@@ -1575,21 +1575,33 @@ function openvpn_resync_all($interface = "") {
 		log_error(gettext("Resyncing OpenVPN instances."));
 	}
 
+	// Check if OpenVPN clients and servers require a resync.
 	if (is_array($config['openvpn']['openvpn-server'])) {
 		foreach ($config['openvpn']['openvpn-server'] as & $settings) {
-			if ($interface <> "" && $interface != $settings['interface']) {
-				continue;
+			$mode_id = "server" . $settings['vpnid'];
+			$fpath = "{$g['varetc_path']}/openvpn/{$mode_id}.interface";
+			$gw_group_change = FALSE;
+			$ip_change = ($interface === $settings['interface']);
```

Comment:
This check is reversed, `$ip_change` should be true when `$interface` differs from `$settings['interface']`.

Reply:
@rbgarga I think you may be mistaken. The code prior to this change will NOT resync when `$interface != $settings['interface']` via use of a continue statement. The change above is syntactic sugar because when `$interface === $settings['interface']` a re-sync occurs via marking the `$ip_change` variable as `TRUE`.

More fundamentally, it's pretty clear that when an interface IP changes that an OpenVPN instance is bound to, a resync must occur...

Reply:
actually I believe we are both mistaken here. I was re-reading the code and when $interface is not empty it means an interface was passed to the function and only tunnels configured to use that interface will be resync'd. I believe that if statement with a continue for the case where interface is not empty and differs from tunnel should be kept as it is today

Reply:
@jwsi You are right and I need more coffee :)